
--- buildroot/package/xenomai/Config.in.orig
+++ buildroot/package/xenomai/Config.in
@@ -3,7 +3,8 @@ config BR2_PACKAGE_XENOMAI_COBALT_ARCH_SUPPORTS
 	default y if BR2_i386
 	default y if BR2_x86_64
 	default y if BR2_powerpc
-	default y if BR2_arm && !BR2_ARM_CPU_ARMV7M && !BR2_ARM_CPU_ARMV8A
+	default y if BR2_aarch64
+	default y if BR2_arm && !BR2_ARM_CPU_ARMV7M
 
 comment "xenomai needs a glibc or uClibc toolchain w/ threads"
 	depends on BR2_USE_MMU
