diff --git a/lib/cobalt/arch/arm/include/asm/xenomai/features.h b/lib/cobalt/arch/arm/include/asm/xenomai/features.h
--- a/lib/cobalt/arch/arm/include/asm/xenomai/features.h
+++ b/lib/cobalt/arch/arm/include/asm/xenomai/features.h
@@ -48,6 +48,10 @@
 #define __LINUX_ARM_ARCH__ 7
 #endif /* armv7 */
 
+#if defined(__ARM_ARCH_8A__)
+#define __LINUX_ARM_ARCH__ 8
+#endif /* armv8 */
+
 #ifndef __LINUX_ARM_ARCH__
 #error "Could not find current ARM architecture"
 #endif
